pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3188,7 +3209,7 @@ public synchronized LongRunningProcessStatus compactionStatus() {
             return LongRunningProcessStatus.forStatus(LongRunningProcessStatus.Status.RUNNING);
         } else {
             try {
-                if (current.join() == COMPACTION_NEVER_RUN) {
+                if (Objects.equals(current.join(), COMPACTION_NEVER_RUN)) {
```

Comment:
There is no need to use `Objects.equals` here. Even if `current.join()` is `null`, `null == COMPACTION_NEVER_RUN` is still a valid expression. The `Objects.equals` method is only used to replace the `x.equals(y)` call in case when `x` is `null`.

Reply:
If current.join() is `-0xfebecffeL` then `current.join() != COMPACTION_NEVER_RUN`.

Reply:
Sorry I don't get it. When does `current.join()` return `-0xfebecffeL`?

In addition,

```java
    public static boolean equals(Object a, Object b) {
        return (a == b) || (a != null && a.equals(b));
    }
```

It's equivalent with `a == b` when `a` and `b` are `Long`.